### PR TITLE
Desktop: add note about rotated key, chrome-sandbox instructions

### PR DESCRIPTION
**Pull request by mvitale1989**
#### Summary

This PR tackles two issues of the Desktop app installation instructions:
- https://github.com/mattermost/docs/issues/7229: add a note on what to do if you have the expired GPG repo key on your system. We had updated the docs for server and omnibus for as part of [this PR](https://github.com/mattermost/docs/pull/7050), but didn't update them for Desktop.
- https://github.com/mattermost/docs/issues/7230: add instructions required to have the mattermost-desktop tarball run on Ubuntu Noble.

Regarding the second issue: this seems to be a widespread issue with electron-builder apps, e.g. see [here](https://youtrack.jetbrains.com/issue/IJPL-59368/IDE-crashes-due-to-chrome-sandbox-is-owned-by-root-and-has-mode-error-when-IDE-is-launching-the-JCEF-in-a-sandbox) or [here](https://github.com/NixOS/nixpkgs/issues/308128). The root cause is the following:
- `chrome-sandbox` needs to be owned by `root` and to have the SETUID bit enabled. If not, AppArmor will deny some operations on Ubuntu Noble (but not on previous Ubuntu releases).
- When electron-builder creates the archive, it gives `root` ownership to all the files in the archive (can be verified with `tar -tv`)
- If the electron-builder tar extraction is done by a non-root user, all of the file will belong to that user instead, including `chrome-sandbox`. This also drops the setuid bit in the process.

The above points mean that with our current installation instructions don't work on Ubuntu Noble, and we must explicitly address the `chrome-sandbox` file permissions when using the tarball.

Note that an alternative solution to what I documented is requesting to extract the whole tarball as root user, e.g. to `/opt/mattermost-desktop/`. cc @devinbinnie please let me know if you have any feedback or more insight on this, thanks!

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-7945
https://mattermost.atlassian.net/browse/CLD-7949

### CODE REVIEW

File: source/collaborate/install-desktop-app.rst [hunk 1]
```diff
@@ -85,6 +85,10 @@ You can download the `desktop app <https://mattermost.com/apps/>`_ directly from
 
   Both a ``.deb`` package (Beta), and an official APT repository is available for Debian 9 and for Ubuntu releases 20.04 LTS or later. Automatic app updates are supported and enabled. When a new version of the desktop app is released, your app updates automatically.
 
+  .. important::
+
+    The GPG public key has changed. If you had previously setup the repository on your system, you'll need to download the new key: you can set the ``UPDATE_GPG_KEY=yes`` environment variable when running the setup script, to configure it to overwrite the previous key on your system with the new one. E.g. the first step of the instructions would become ``curl -fsS -o- https://deb.packages.mattermost.com/setup-repo.sh | sudo UPDATE_GPG_KEY=yes bash``. Note that depending on your setup, additional steps may also be required, particularly for installations that didn't rely on the repository setup script.
```
Comment (cwarnermm) [inline source/collaborate/install-desktop-app.rst:90]:
```suggestion
    The GPG public key has changed. If you had previously set up the repository on your system, you'll need to download the new key. You can set the ``UPDATE_GPG_KEY=yes`` environment variable when running the setup script to configure it to overwrite the previous key on your system with the new one. The first step of installation then becomes: ``curl -fsS -o- https://deb.packages.mattermost.com/setup-repo.sh | sudo UPDATE_GPG_KEY=yes bash``. Depending on your setup, additional steps may also be required, particularly for installations that don't rely on the repository setup script.
```

**Review by cwarnermm — APPROVED** [review]
Thank you, @mvitale1989! I've proposed editorial updates inline. Approving for expediency.

**Review by phoinixgrr — APPROVED** [review]

File: source/collaborate/install-desktop-app.rst (hunk 1 above)
Comment (mvitale1989) [reply source/collaborate/install-desktop-app.rst]: Thanks for the improvements Carrie :tada:

**Comment by mvitale1989** [comment]
I'll merge this as is, if there's more feedback on this later we can revisit the instructions.

**Pull request merged**